apps/prairielearn/src/pages/administratorCourseRequests/administratorCourseRequests.js

```diff
@@ -18,8 +25,35 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     res.locals.coursesRoot = config.coursesRoot;
-    const result = await sqldb.queryOneRowAsync(sql.get_requests, []);
-    _.assign(res.locals, result.rows[0]);
+    const result = await sqldb.queryRow(
+      sql.get_requests,
+      [],
+      z.object({
+        institutions: z.array(InstitutionSchema),
+        course_requests: z.array(
+          CourseRequestSchema.extend({
+            user_name: z.string(),
+            created_at: DateFromISOString.optional(),
+            approved_by: z.string().optional(),
```

Comment:
Why is this marked `optional` and not `nullable`?

Reply:
The query was not returning a value for `approved_by` and I needed to remove this from the CourseRequestSchema or mark that the value could be undefined. I have now written a custom `CourseRequestRowSchema` that now actually matches the results of the query.